code/game/machinery/turrets/subtypes/misc.dm

```diff
@@ -0,0 +1,116 @@
+	installation = /obj/item/gun/energy/xray // For the armor pen.
+	integrity = 250 // Since lasers do 40 each.
+	integrity_max = 250
+
+/obj/machinery/porta_turret/ai_defense/emp_act(severity)
+	if(prob(33)) // One in three chance to resist an EMP.  This is significant if an AoE EMP is involved against multiple turrets.
+		return
+	..()
+
+/datum/category_item/catalogue/anomalous/precursor_a/alien_turret
+	name = "Precursor Alpha Object - Turrets"
+	desc = "An autonomous defense turret created by unknown ancient aliens. It utilizes an \
+	integrated laser projector to harm, firing a cyan beam at the target. The signal processing \
+	of this mechanism appears to be radically different to conventional electronics used by modern \
+	technology, which appears to be much less susceptible to external electromagnetic influences.\
+	<br><br>\
+	This makes the turret be very resistant to the effects of an EM pulse. It is unknown if whatever \
+	species that built the turret had intended for it to have that quality, or if it was an incidental \
+	quirk of how they designed their electronics."
+	value = CATALOGUER_REWARD_MEDIUM
+
+/obj/machinery/porta_turret/alien // The kind used on the UFO submap.
+	name = "interior anti-boarding turret"
+	desc = "A very tough looking turret made by alien hands."
+	catalogue_data = list(/datum/category_item/catalogue/anomalous/precursor_a/alien_turret)
+	icon_state = "turret_cover_alien"
+	req_one_access = list(ACCESS_FACTION_ALIEN)
+	installation = /obj/item/gun/energy/alien
+	enabled = TRUE
+	lethal = TRUE
+	ailock = TRUE
+	check_all = TRUE
+	integrity = 250 // Similar to the AI turrets.
+	integrity_max = 250
+	turret_type = "alien"
+
+/obj/machinery/porta_turret/alien/emp_act(severity) // This is overrided to give an EMP resistance as well as avoid scambling the turret settings.
+	if(prob(75)) // Superior alien technology, I guess.
+		return
+	enabled = FALSE
+	spawn(rand(1 MINUTE, 2 MINUTES))
+		if(!enabled)
+			enabled = TRUE
+
+/obj/machinery/porta_turret/alien/destroyed // Turrets that are already dead, to act as a warning of what the rest of the submap contains.
+	name = "broken interior anti-boarding turret"
+	desc = "A very tough looking turret made by alien hands. This one looks destroyed, thankfully."
+	icon_state = "destroyed_target_prism_alien"
+	machine_stat = BROKEN
+	can_salvage = FALSE // So you need to actually kill a turret to get the alien gun.
+
+/obj/machinery/porta_turret/industrial
+	name = "industrial turret"
+	desc = "This variant appears to be much more rugged."
+	req_one_access = list(ACCESS_COMMAND_BRIDGE)
+	icon_state = "turret_cover_industrial"
+	installation = /obj/item/gun/energy/phasegun
+	integrity = 200
+	integrity_max = 200
+	turret_type = "industrial"
+
+/obj/machinery/porta_turret/industrial/bullet_act(obj/projectile/Proj)
+	. = ..()
+	if(enabled)
+		if(!attacked && !emagged)
+			attacked = TRUE
+			spawn()
+				sleep(60)
```

Comment:
can we use seconds here instead of passing in deciseconds

Reply:
this is a moved segment but yea sure